magma/compile_guard.py

```diff
@@ -95,6 +94,21 @@ def _new_port_name(self) -> str:
         return f"port_{next(self._num_ports)}"
 
 
+def get_compile_guard_data(
+        inst_or_defn: Union[AnonymousCircuitType, CircuitKind]
+) -> Optional[Dict]:
```

Comment:
Could consider moving this API to the circuit objects and have them implement it.